### PR TITLE
Add 3 proposals to babel config

### PR DESCRIPTION
**Pull request by pnowak**
### Context

I needed to add `eslint-plugin-babel` and update `babel-eslint` `eslint` to work with these three experimental features. And I didn't have time to implement `private methods`, I only changed our `privatePool` to private fields.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #6308

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation.

### CODE REVIEW

**Comment by pnowak** [comment]
We have a blocker - https://github.com/babel/babel-eslint/issues/749 - in `private method`.  So we need to wait until this problem will be resolved.

**Comment by pnowak** [comment]
@budnix After review I'll remove in `copyPaste` file every change - it has only demonstrative and tests purpose.

File: src/plugins/copyPaste/copyPaste.js [hunk 1]
```diff
@@ -125,15 +131,18 @@ class CopyPaste extends BasePlugin {
     if (this.enabled) {
       return;
     }
-    const settings = this.hot.getSettings();
 
-    this.#isFragmentSelectionEnabled = settings.fragmentSelection;
+    const { copyPaste: settings, fragmentSelection } = this.hot.getSettings();
```
Comment (wojciechczerniak) [inline src/plugins/copyPaste/copyPaste.js]: Renaming `copyPaste` to `settings` is confusing. My first thought was that `pasteMode` is not in our main settings object.

Comment (pnowak) [reply src/plugins/copyPaste/copyPaste.js]: This is not my code, it is on `develop` branch.
As I said above, `copyPaste` is only for only demonstrative purpose. And I'll remove all changes after review.

Comment (wojciechczerniak) [reply src/plugins/copyPaste/copyPaste.js]: Oops, sorry. Didn't noticed that. I had a highlighted change but you're right, that was `develop` merge.

I don't think you should remove the changes in `copyPaste`. New operators should stay as a part of the code base and be a living proof that the babel plugins are working correctly.

Comment (budnix) [reply src/plugins/copyPaste/copyPaste.js]: I think that you can remove all code not related to this PR before review. The description of this PR is enough descriptive to handle a review. Moreover, I can look into the code with example syntax by analyzing previous commits.

Comment (wojciechczerniak) [reply src/plugins/copyPaste/copyPaste.js]: Right, I'm not the reviewer 🤐 😄 👍

**Comment by wojciechczerniak** [comment]
@pnowak What's our status on this?

**Comment by pnowak** [comment]
I'm waiting for @budnix to upgrade the node version in the documentation.

**Comment by budnix** [comment]
Ok! I've upgraded node version, and we're ready for this PR. However, it's been a while since the last commit to this branch. Can I ask you @pnowak to check dependencies to see if they need updates?

**Comment by budnix** [comment]
@pnowak Please sync with develop branch, here we have some issues with npm audit. Some of the was fixed on develop branch.

**Comment by pnowak** [comment]
It seems that we can remove both plugins - `@babel/plugin-proposal-class-properties` and `plugin-proposal-nullish-coalescing-operator` - https://babeljs.io/blog/2020/01/11/7.8.0 
And update babel version to 7.8.0. Isn't it?

**Comment by budnix** [comment]
> It seems that we can remove both plugins - `plugin-proposal-optional-chaining` and `plugin-proposal-nullish-coalescing-operator` - https://babeljs.io/blog/2020/01/11/7.8.0
> And update babel version to 7.8.0. Isn't it?

Sure! Let's do it. It can be necessary to bump other babel-related modules.

**Pull request merged**